cmd/lncli/commands.go

```diff
@@ -2070,6 +2071,25 @@ func parseChanPoint(s string) (*lnrpc.ChannelPoint, error) {
 	}, nil
 }
 
+// parseTimeLockDelta is expected to get a uint16 type of timeLockDelta,
+// which maximum value is MaxTimeLockDelta.
+func parseTimeLockDelta(timeLockDeltaStr string) (uint16, error) {
+	timeLockDeltaUnCheck, err := strconv.ParseUint(
+		timeLockDeltaStr, 10, 64,
+	)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse time_lock_delta: %s "+
+			"to uint64, err: %v", timeLockDeltaStr, err)
+	}
+
+	if timeLockDeltaUnCheck > uint64(lnd.MaxTimeLockDelta) {
+		return 0, fmt.Errorf("time_lock_delta is too big, "+
+			"max value is %d", lnd.MaxTimeLockDelta)
+	}
```

Comment:
dont think we need this check on this side if it will already be checked on the server side.

Reply:
Perhaps it is better to keep this check, because the code on the server side may not be updated in time.

Reply:
I think it is usually expected that the server updates before the client does? In my mind, "business logic" checks like this should be left to the server. 

The other reason for removing it would be to make this function more generic so that it can be used to check overflows of other parameters.

Reply:
Agree with you. Will remove it.

Reply:
I would've agreed with this, **if** the return value of this function was `uint32`. But given it's `uint16` that can never be larger than the allowed maximum value. Instead a too large value would silently be converted/truncated to an `uint16` value, which might lead to very unexpected behavior.